V2.29.0 k8s 1.33.5 igz 3.7.2

shay79il opened the pull request.

murakami-nec committed:
feat: Add Dockerfile.rocky10

murakami-nec committed:
feat: Add support for RHEL 10 (#57)

- Use python3.12
- Don't need modulemd-tools to create repo

tmurakam committed:
fix: Set umask 022 before downloads (#58)

tmurakam committed:
test: update Vagrantfile for libvirt

tmurakam committed:
feat: Bump kubespray 2.28.1

tmurakam committed:
remove: Drop ubuntu 20.04 support

tmurakam committed:
test: Rename test scripts

tmurakam committed:
docs: update ChangeLog

vainkop committed:
Update get-kubespray.sh to to be able to use git commit hash

mkl262 committed:
feat: Use kubespray 2.29.0

vainkop committed:
Re-add  the last LF code.

tmurakam committed:
Merge pull request #60 from vainkop/main

Update get-kubespray.sh to to be able to use git commit hash

tmurakam committed:
Merge pull request #63 from mkl262/main

feat: Use kubespray 2.29.0

tmurakam committed:
feat: Update requirements.txt for kubespray 2.29.0

tmurakam committed:
remove: Drop support of RHEL 8 family

The RHEL8 support was dropped by Kubespray 2.29.0

tmurakam committed:
feat: Bump nginx 1.28.0 -> 1.29.2

tmurakam committed:
docs: Update ChangeLog

shay79il committed:
bump k8s version to 1.33.5

shay79il committed:
Merge tag 'v2.29.0-0' into v2.29.0_k8s-1.33.5_igz-3.7.2

alexpirogovski commented on igz_files/igz_override.yml.j2:
```diff
@@ -88,3 +88,6 @@ kubelet_config_extra_args:
 kubeadm_ignore_preflight_errors:
   - SystemVerification
 
+# Containerd version compatible with Rocky 8 (GLIBC 2.28)
```
Please specify that this is a downgrade from what is used in upstream Kubespray

shay79il committed:
Add Rocky 8 compatibility patches for Kubespray 2.29.0

Pre-build patches (igz_patches/2.29.0-pre/):
- config.sh.patch: Downgrade containerd 2.1.4 -> 2.0.5 (GLIBC 2.28 compatible)
- download-kubespray-files.patch: Pass containerd version to generate_list.sh
- requirements.patch: Downgrade Ansible 10.7.0 -> 9.13.0 (ansible-core 2.16.x)

Direct file changes (igz_files/igz_override.yml.j2):
- containerd_version: 2.0.5 (Rocky 8 GLIBC 2.28 compatible)
- minimal_ansible_version: 2.15.0 (override Kubespray's 2.17.3 requirement)
- maximal_ansible_version: 2.19.0 (override Kubespray's 2.18.0 limit)

Misc:
- Add .DS_Store to .gitignore

Rocky 8 has GLIBC 2.28. containerd 2.1.4 requires GLIBC 2.34, while
containerd 2.0.5 only requires GLIBC 2.4, making it compatible with Rocky 8.

Configuration aligned with v2.28.0 branch (Python 3.10, Ansible 9.13.0).
Ansible version override uses variables instead of patching Kubespray files.

alexpirogovski reviewed (commented):
LGTM
